app/api/services/informationextraction/InformationExtraction.ts

```diff
@@ -0,0 +1,448 @@
+
+    if (files.length === 0) {
+      emitToTenant(tenants.current().name, 'ix_model_status', property, 'ready', 'Completed');
+      return;
+    }
+    await this.sendMaterials(files, property, serviceUrl, 'prediction_data');
+  };
+
+  getFiles = async (
+    templates: ObjectIdSchema[],
+    property: string,
+    toTrain: boolean = true
+  ): Promise<FileWithAggregation[]> => {
+    let agg: any = [
+      {
+        $match: {
+          type: 'document',
+          filename: { $exists: true },
+        },
+      },
+      {
+        $lookup: {
+          from: 'entities',
+          localField: 'entity',
+          foreignField: 'sharedId',
+          as: 'entityData',
+        },
+      },
+      {
+        $lookup: {
+          from: 'segmentations',
+          localField: '_id',
+          foreignField: 'fileID',
+          as: 'segmentation',
+        },
+      },
+      {
+        $match: {
+          'entityData.template': { $in: templates },
+          segmentation: { $ne: [] },
+        },
+      },
+    ];
+
+    if (toTrain) {
+      agg.push({
+        $match: {
+          'extractedMetadata.name': property,
+        },
+      });
+    }
+
+    if (!toTrain) {
+      const [currentModel] = await IXModelsModel.get({ propertyName: property });
+      agg = agg.concat([
+        {
+          $lookup: {
+            from: 'ixsuggestions',
+            localField: 'entity',
+            foreignField: 'entityId',
+            as: 'suggestions',
+          },
+        },
+        {
+          $match: {
+            $or: [
+              {
+                suggestions: {
+                  $size: 0,
+                },
+              },
+              {
+                'suggestions.date': { $lte: currentModel.creationDate },
+                'suggestions.propertyName': property,
+              },
+            ],
+          },
+        },
+        {
+          $limit: this.batchSize,
+        },
+      ]);
+    }
+    return filesModel.db.aggregate(agg);
+  };
+
+  trainModel = async (property: string) => {
+    const templates: ObjectIdSchema[] = await this.getTemplatesWithProperty(property);
+    const serviceUrl = await this.serviceUrl();
+    const materialsSent = await this.materialsForModel(templates, property, serviceUrl);
+    if (!materialsSent) {
+      return { status: 'error', message: 'No labeled data' };
+    }
+
+    await this.taskManager.startTask({
+      task: 'create_model',
+      tenant: tenants.current().name,
+      params: { property_name: property },
+    });
+
+    await this.saveModelProcess(property);
+    return { status: 'processing_model', message: 'Training model' };
+  };
+
+  status = async (property: string) => {
+    const [currentModel] = await IXModelsModel.get({
+      propertyName: property,
+      status: 'processing',
+    });
+
+    const [suggestion] = await IXSuggestionsModel.get({
+      propertyName: property,
+      status: 'processing',
+    });
+
+    if (currentModel) {
```

Comment:
Maybe move this between the two requests above, since we don't need `suggestion` if we return from this first "if".

Reply:
done